### PR TITLE
Add support for running 'pick' on sequences of spectrum images.

### PR DESCRIPTION
**Pull request by Brow71189**
This is in preparation for and updated `MultiAcquire` that will allow users to acquire sequences of spectrum images.

### CODE REVIEW

**Comment by Brow71189** [comment]
For some reason the tests are always run with numpy 1.15 for python 3.6 and 3.7. They fail because the `where` keyword argument was introduced in numpy 1.17 (https://docs.scipy.org/doc/numpy/reference/generated/numpy.sum.html#numpy.sum)

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
     long_description=open("README.rst").read(),
     url="https://github.com/nion-software/niondata",
     packages=["nion.data", "nion.data.test"],
-    install_requires=['scipy', 'numpy>=1.16', 'nionutils'],
+    install_requires=['scipy', 'numpy>=1.17', 'nionutils'],
```
Comment (cmeyer) [inline setup.py]: Also requires a change in `test-requirements.txt`. I just made a patch to require numpy>=1.16. Can you include the update to 1.17 in `test-requirements.txt` also in your PR?

**Comment by cmeyer** [comment]
I think you need to pull upstream and rebase.

**Comment by Brow71189** [comment]
Yes of course I forgot to merge first. No it's done and the checks should pass.

**Comment by cmeyer** [comment]
You can also squash when you force-push.

**Comment by Brow71189** [comment]
I merged the changes an made a new commit. Didn't want to force-push.

**Comment by Brow71189** [comment]
Would you prefer me to squash the commits and force-push?

File: nion/data/Core.py [hunk 2]
```diff
@@ -1313,23 +1330,33 @@ def function_average_region(data_and_metadata: DataAndMetadata.DataAndMetadata,
...
-    assert len(data_and_metadata.dimensional_shape) == 3
+    if data_and_metadata.is_sequence:
+        assert len(data_and_metadata.dimensional_shape) == 4
+    else:
+        assert len(data_and_metadata.dimensional_shape) == 3
     assert len(mask_data_and_metadata.dimensional_shape) == 2
 
     data = data_and_metadata.data
-    mask_data = mask_data_and_metadata.data
+    mask_data = mask_data_and_metadata.data.astype(numpy.bool)
```
Comment (cmeyer) [inline nion/data/Core.py]: Why is this needed? This slows down the use of mask data. Can this be a separate patch with a test if it is triggering a problem somewhere?

**Comment by cmeyer** [comment]
I think to overcome the conflict you will have to rebase and that will require a force-push. Squashing is optional, but it looks like the commits so far have been minor edits.

File: nion/data/Core.py (hunk 2 above)
Comment (Brow71189) [reply nion/data/Core.py]: This is because the `where` keyword argument requires a boolean array. It just causes an exception if the mask is not of type `bool`.
Overall this will still be faster than the previous version which multiplied the whole multidimensional array with the mask. Now, only the rather small (2D) mask will be converted and numpy will only consider those parts of the big array for the sum that are actually needed.

File: CHANGES.rst [hunk 3]
```diff
@@ -17,6 +17,7 @@ Changelog (niondata)
 0.13.8 (2019-10-24)
 -------------------
 - Added optional registration area bounds to align and register functions.
+- Make pick functions work for sequences of spectrum images.
```
Comment (cmeyer) [inline CHANGES.rst]: This should go into a new 0.13.9 (UNRELEASED) section.

File: test-requirements.txt [hunk 4]
```diff
@@ -5,6 +5,6 @@
 
 -e git://github.com/nion-software/nionutils.git#egg=nionutils
 
-numpy>=1.16
+numpy>=1.17
```
Comment (cmeyer) [inline test-requirements.txt]: This PR needs to be rebased and pushed again to avoid this conflict.

**Comment by Brow71189** [comment]
I implemented all requested changes, rebased and pushed.

**Comment by cmeyer** [comment]
This is still showing a branch conflict (look at the last message on this issue's page). To fully rebase you need to pull from upstream, rebase from master (17e2d09c9faf09960805f2d53e5345b7b45b1bf0 at the moment), then force push to this branch.

**Comment by cmeyer** [comment]
Merged. 141aa5dc34b8b73f96e3c91b9f337b702633525e
